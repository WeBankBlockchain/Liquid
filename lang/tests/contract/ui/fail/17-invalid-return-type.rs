use liquid_lang as liquid;

#[derive(Clone, Copy)]
pub struct I(i32);

#[liquid::contract]
mod noop {
    use super::*;

    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) -> I {}
    }
}

fn main() {}
