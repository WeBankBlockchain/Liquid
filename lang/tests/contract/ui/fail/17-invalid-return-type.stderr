error[E0277]: the trait bound `I: You_Should_Use_An_Valid_Output_Type` is not satisfied
  --> $DIR/17-invalid-return-type.rs:17:31
   |
17 |         pub fn noop(&self) -> I {}
   |                               ^ the trait `You_Should_Use_An_Valid_Output_Type` is not implemented for `I`
